### ISSUE TITLE
[Task]: Mark some add-ons as Partner add-ons in the reviewer tools

### ISSUE THREAD

**Issue by abyrne-moz** [mozilla/addons#14986]
### Description

Any “negative” impact to either the listing or users on either Notable or Partner addons, such as delisting, disabling or blocklist, needs a second checkpoint to ensure additional process is handled prior to the action taking affect.  The intention of the checkpoint is to verify communications have been handled, the developer is aware, and we are taking precautions in how end users are impacted.  

Until further automated handling of the workflow is available, reviewers will need to notify operations that the checkpoint verification is needed.  To do this, the reviewers need to be aware that the addon they are reviewing is Notable and/or Partner related.

### Acceptance Criteria

```[tasklist]
  ### Milestones/checkpoints
  - [ ] A tag of some kind is shown in reviewer tools indicating Notable and Partner status if those are set
  - [ ] Partner status is per-addon [not developer or addon version]
  - [ ] Partner status should not be publicly exposed through any means, API or otherwise
  - [ ] Partner status should appear in reviewer tools independently of any other flags, the partner status should not disappear if other flags, such as promoted flags, are present
  - [ ] AMO does not display partner status via the API
  - [ ] Partner addons may be listed and/or unlisted
  - [ ] Partner status is managed through django admin
  - [ ] List of Partner addons is managed or provided by Business Development
  - [ ] This information should be exposed to Cinder moderators
```

### Checks

- [X] If the issue is ready to work on, I have removed the "needs:info" label.
- [X] If I have identified that the work is specific to a repository, I have removed "repository:addons-server" or "repository:addons-frontend"

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1011)

**Comment by diox** [comment]
Question that needs to be asked: do we want pre-review on Partner add-ons ? If so, this makes the implementation of that ticket more involved, we would lean towards re-using promoted add-on system in that case.

However to do that, promoted groups would need to be refactored to allow multiple groups/promoted states for a given add-on.

**Comment by diox** [comment]
Per discussion in standup, it looks like we do want pre-review for partner.

Plan is to:
- Have a spike to investigate having multiple promoted groups for an add-on
- If that turns out to be easy to do, implement partner add-ons that way with the refactor
- If not, we could still implement partner add-ons with the promoted system without refactor but we'd have to do it eventually anyway...

**Comment by ioanarusiczki** [comment]
@chrstinalin 

>A tag of some kind is shown in reviewer tools indicating Notable and Partner status if those are set

Yes, I see the notable status in [rev tools ](https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/636957). There's no partner status available in dev hub (but this could be expected ❓)

> Partner status is per-addon [not developer or addon version]

Yes, this can be noticed in admin tools on a [partner addon](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/discoveryaddon/636957/change/) with multiple versions in Promoted addon versions section -> there are no versions here as for other promotions (recommended for example) ✅

>Partner status should not be publicly exposed through any means, API or otherwise

There is not frontend badge and in the API, for example on [this request]( https://addons-dev.allizom.org/api/v5/addons/addon/636957/) I get  `"promoted": []` ✅

>Partner status should appear in reviewer tools independently of any other flags, the partner status should not disappear if other flags, such as promoted flags, are present

I couldn't test this yet. Adding "Recommeded" or another promotion is not possible now. ❌

>Partner addons may be listed and/or unlisted

I could add listed and unlisted extensions. Also dictionaries, lang packs or themes can be Partner addons. ✅

>Partner status is managed through django admin

This has been tested from [discoveryaddon](https://addons-internal.dev.mozaws.net/en-US/admin/models/discovery/discoveryaddon/) on AMO dev ✅

>This information should be exposed to Cinder moderators

I could see in Cinder on the left in [Details](https://stage.cinder.nonprod.webservices.mozgcp.net/job/8cf9c99d-1794-4275-b0ba-e3ee05cfa2ce) that's "Promoted Partner" ✅

Partner addons with DSA:

- rejecting a Partner addon from rev tools is held for rejection ✅
- disabling a Partner addon from Cinder is held for disable ✅
- user ban with a Partner addon submitted is held in 2nd level approval queue ✅
- a dev reply on a Partner addon is deleted , according to [this document](https://docs.google.com/document/d/1kOPhRdTuodEjjQSyLr1EB12bFQCCpp8SKtFR4wUXoIk/edit?tab=t.0#heading=h.9ko9o62hoolg) , decision should be held (going to file a followup) ✅

Detailed [notes/testing](https://docs.google.com/document/d/1UBIA_1QxIn_wFzhPTnbQITk4qn2t2z-1dxo_PuOymzc/edit?tab=t.0) (that's stuffs that I'm interested in, but it's good to have it attached here).

**Comment by ioanarusiczki** [comment]
Ok, thanks @chrstinalin I'll mark this verified then we'll see what happens after the last part lands.
